### DISCUSSION TITLE
Media/consumer keys not working in usb host mode

### DISCUSSION THREAD

**Discussion by LimeProgramming** [hathach/tinyusb#1704] [Q&A]
Hey.

I've recently tinkering with a project where the pi pico converts usb keyboard to a ps/2 keyboard. While the main keyboard works, the media keys (Like volume or open calculator button) are not triggering the hid report received callback. 

I see in example code in process generic report notes about hid usage pages and it implies that the consumer keys should trigger the callback to get to the process generic report but they are not. 

I also see that if I'm using the pico as usb keyboard, I could send said consumer keycodes.

Does the USB host not support consumer keys? or am I missing something?

I tried using the host examples given but the callback isn't triggered for them either.

**Comment by hathach** [comment]
for simplicity, tinyusb host will start with BOOT mode. To support consumer key, you will need to switch protocol to REPORT mode, and also need to parse report descriptor to know which report and what layout it uses for consumer key.
